workers/unity/Assets/Tests/EditmodeTests/Correctness/Core/CommandReceivedResponseTests.cs

```diff
@@ -0,0 +1,73 @@
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.TestUtils;
+using Improbable.Worker.CInterop;
+using NUnit.Framework;
+using Playground;
+
+namespace Improbable.Gdk.EditmodeTests.Core
+{
+    public class CommandReceivedResponseTests : MockBase
+    {
+        private const long EntityId = 101;
+        private const long LaunchedId = 102;
+
+        [Test]
+        public void ReceivedResponse_isNull_when_Response_is_not_received()
+        {
+            World.Step(world =>
+            {
+                world.Connection.CreateEntity(EntityId, GetTemplate());
+            }).Step(world => world.GetSystem<CommandSystem>().SendCommand(GetRequest())).Step((world, id) =>
+            {
+                Assert.IsFalse(world.GetSystem<CommandSystem>().GetResponse<Launcher.LaunchEntity.ReceivedResponse>(id).HasValue);
+            });
+        }
+
+        [Test]
+        public void ReceivedResponse_isNotNull_when_Response_is_received()
+        {
+            World.Step(world =>
+            {
+                world.Connection.CreateEntity(EntityId, GetTemplate());
+            }).Step(world =>
+            {
+                var id = world.GetSystem<CommandSystem>().SendCommand(GetRequest());
+                world.Connection.GenerateResponse<Launcher.LaunchEntity.Request, Launcher.LaunchEntity.ReceivedResponse>(id, ResponseGenerator);
+                return id;
+            }).Step((world, id) =>
+            {
+                Assert.IsTrue(world.GetSystem<CommandSystem>().GetResponse<Launcher.LaunchEntity.ReceivedResponse>(id).HasValue);
```

Comment:
It may be worth adding an `Assert` against the contents of the response to ensure its the one that we responded with rather than the API returning something else.